File: Lidar Test Code/java/src/main/java/frc/robot/commands/StartStop.java
package frc.robot.commands;

import edu.wpi.first.wpilibj2.command.CommandBase;
import frc.robot.RobotContainer;

import frc.robot.subsystems.LidarTest;
import frc.robot.gamepad.OI;

public class StartStop extends CommandBase
{
    private static final LidarTest lidar = RobotContainer.lidar;
    private static final OI oi = RobotContainer.oi;


    public StartStop ()
    {
        addRequirements(lidar);
    }


    @Override
    public void initialize()
    {}

    @Override
    public void execute()
    {
        // Simple way to show starting and stopping of lidar
        if (oi.getDriveXButton())
        {
            lidar.startScan();
        }
        else if (oi.getDriveYButton())
        {
            lidar.stopScan();
        }
    }

    @Override
    public void end (boolean interrupted)
    {
        lidar.stopScan();
    }

    @Override
    public boolean isFinished()
    {
        return false;
    }
}

File: Lidar Test Code/cpp/src/main/cpp/commands/StartStop.cpp
#include "commands/StartStop.h"

StartStop::StartStop(LidarTest* lidar, OI* oi)
{
    AddRequirements({lidar, oi});
    this->lidar = lidar;
    this->oi = oi;
}

void StartStop::Initialize(){}

void StartStop::Execute()
{
    // Simple way to show stopping and starting lidar
    if (oi->GetDriveXButton())
    {
        std::cout<<"Start Lidar\n";
        lidar->StartLidar();
    }
    else if (oi->GetDriveTriangleButton())
    {
        std::cout<<"Stop Lidar\n";
        lidar->StopLidar();
    }
}

void StartStop::End(bool interrupted)
{
    lidar->StopLidar();
}

bool StartStop::IsFinished()
{
    return false;
}